### ISSUE TITLE
linter: Implement new parameters to control line-index and column-index conversion to handle zero-based and one-based conventions

### ISSUE THREAD

**Issue by Makman2** [coala/coala#5346]
Many tools have different conventions regarding what is the first line or column. Some emit `0`, some emit `1`.

Acceptance criteria:
- `@linter` accepts argument(s) that controls the conversion of zero-based line/column indexes. When enabled, zero-based indexes are incremented by 1 to match coala's (one-based) convention. The user is able to control column- and line-conversion individually.
- The new parameter is documented in our API and in our tutorials and guides (namely `Writing_Linter_Bears.rst`).
- Tests exist for the new parameter.

~~Notes:
From my experience I would recommend to have only one parameter that controls the conversion for both, lines and columns at once.~~

**Comment by jayvdb** [comment]
> From my experience I would recommend to have only one parameter that controls the conversion for both, lines and columns at once.

Why?  This has been implemented in the PR, and it looks cumbersome.

Without seeing your recommendation, I had suggested switching to two bools.  80/20 rule.  That makes most cases easy, and more complicated cases will need to implement their own method to yield the Results.

In almost all cases, the linter has intentionally chosen a particular convention, and the choice of column convention is not the same as row convention.

A scenario we might want to plan for is where a linter gets it completely wrong, with different offset conventions for different results.  IMO we might decide to just discard their rows and/or cols in that case, which can be done by amending the regex or similar (and raise bugs upstream, and wait for fixes).
